### PR TITLE
Homework#8

### PR DESCRIPTION
**Pull request by yeargor**

### CODE REVIEW

File: src/main/java/Homework8/config/JDBCConfig.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    public static SessionFactory getSessionFactory(){
+        return new Configuration()
+                .configure(HIBERNATE_CFG_XML)
+                .addAnnotatedClass(User.class)
+                .addAnnotatedClass(Ticket.class)
+                .buildSessionFactory();
+    }
+
+    public static Session getSession(){ return getSessionFactory().getCurrentSession(); }
```
Comment (Alexmansar) [inline src/main/java/Homework8/config/JDBCConfig.java:21]: fix formatting

File: src/main/java/Homework8/config/JDBCConfig.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import Homework8.model.Ticket;
+import Homework8.model.User;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.cfg.Configuration;
+
+public class JDBCConfig {
+
+    public static final String HIBERNATE_CFG_XML = "hibernate.cfg.xml";
```
Comment (Alexmansar) [inline src/main/java/Homework8/config/JDBCConfig.java:11]: couldbe private

File: src/main/java/Homework8/model/Ticket.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    @ToString.Exclude
+    private User user;
+    @Enumerated(EnumType.STRING)
+    TicketType ticketType;
+    Date creationDate;
+
+    public Ticket(User user, TicketType ticketType, Date creationDate) {
+        this.user = user;
+        this.ticketType = ticketType;
+        this.creationDate = creationDate;
```
Comment (Alexmansar) [inline src/main/java/Homework8/model/Ticket.java:32]: also, you can crete this one in the constructor

File: src/main/java/Homework8/model/User.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    int id;
+    String name;
+    Date creationDate;
+    @OneToMany(mappedBy = "user", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+    private Set<Ticket> tickets;
+
+    public User(String name, Date creationDate) {
+        this.name = name;
+        this.creationDate = creationDate;
```
Comment (Alexmansar) [inline src/main/java/Homework8/model/User.java:29]: same

File: src/main/java/Homework8/repository/TicketRepository.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        }catch (Exception e){
+            System.out.println(e.getMessage());
+        }
+    }
+
+    public Ticket getTicketById(int id){
+        try(Session session = JDBCConfig.getSession()){
+            session.beginTransaction();
+            Ticket ticket = session.get(Ticket.class, id);
+            return ticket;
```
Comment (Alexmansar) [inline src/main/java/Homework8/repository/TicketRepository.java:28]: could be just 
return session.get(Ticket.class, id);

File: src/main/java/Homework8/repository/UserRepository.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        }catch (Exception e){
+            System.out.println(e.getMessage());
+        }
+    }
+
+    public User getUser(int userId){
+        try(Session session = JDBCConfig.getSession()){
+            session.beginTransaction();
+            User user = session.get(User.class, userId);
+            return user;
```
Comment (Alexmansar) [inline src/main/java/Homework8/repository/UserRepository.java:27]: same

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@yeargor HI)) in general good job)) just small comments))

File: src/main/java/Homework8/model/Ticket.java (hunk 3 above)
Comment (yeargor) [reply src/main/java/Homework8/model/Ticket.java:32]: Could you please explain what I need to do? I dont understand a bit, sorry

Comment (Alexmansar) [reply src/main/java/Homework8/model/Ticket.java:32]: this. this.creationDate  = new Date(System.currentTimeMillis()
and remove from the constructor parameters

**Review by Alexmansar — APPROVED** [review]
@yeargor Hi)) please, merge it)) good job

**Pull request merged**